example-dotnet/Program.cs

```diff
@@ -11,8 +22,134 @@ static void Main(string[] args)
+	    // out, and we instead opted to include a simple access to
+	    // a constant in the library. This is to exemplify the
+	    // negative case of breaking update over different sets of
+	    // frameworks between the current version and the
+	    // recommended update.
+
+            // Console.WriteLine("Unzipping example ZIP file using SharpZipLib");
+            // new FastZip().ExtractZip(@"resources" + Path.DirectorySeparatorChar + "example.zip", @".", null);
+
+            // Here we simply retrieve a constant within SharpZipLib
+            // to avoid method call.
+            Console.WriteLine("SharpZipLib ZipConstants.VersionMadeBy = " + ZipConstants.VersionMadeBy);
+
+            // Usage of DotNetZip API
+      
+            // Unzip the file to memory using DotNetZip
+            // (vulnerable). This may be reaching methods affected by
+            // the update. The zip extraction API of DotNetZip 1.10.0
+            // has been reported to have an 'Arbitrary File Write'
+            // (CVE-2018-1002205) vulnerability. The vulnerability is
+            // fixed in 1.11.0. The used API is compatible with the
+            // fixed version. SourceClear agent recommends updating
+            // 1.10.0 to 1.11.0. Their nuget packages include the same
+            // net20 framework. This exemplifies the possible positive
+            // case of breaking update over the same framework between
+            // the current version (1.10.0) and the recommended update
+            // to resolve the vulnerability.
+
+	    // This is necesssary to prevent DotNetZip complaining
+	    // about IBM437 encoding.
+	    Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+	    
+            Console.WriteLine("Unzipping example ZIP file using DotNetZip");
+
+            // From https://dotnetfiddle.net/U1pPeD
+            using (Ionic.Zip.ZipFile zip = Ionic.Zip.ZipFile.Read(@"resources" + Path.DirectorySeparatorChar + "example.zip")) {
+                Console.WriteLine("Files found in zip:");
+                foreach (Ionic.Zip.ZipEntry e in zip) {
+                    Console.WriteLine(e.FileName);
+                }
+            }
+
+            // Usage of SharpCompress API
+
+	    // This uses vulnerable zip extraction API of
+	    // SharpCompress 0.17.1. This library is reported to have
+	    // an 'Arbitrary File Write' (CVE-2018-1002206)
+	    // vulnerability. The vulnerability is fixed in 0.21.0. In
+	    // addition, we also use the class
+	    // `ReadOnlyAppendingStream` which does not exist in
+	    // 0.21.0. 0.17.1 includes net35, net45, netstandard1.0,
+	    // and netstandard1.3 framework versions, whereas 0.21.0
+	    // in addition includes netstandard2.0 framework
+	    // version. This case demonstrates a definite breaking
+	    // update from 0.17.1 to 0.21.0 over different sets of
+	    // frameworks between the used version 0.17.1 and the
+	    // recommended update.

```

Comment:
Maybe we could separate this paragraph and put the explanations near their respective uses.

Reply:
This has been summarized in a table.